### PR TITLE
Fix autofixer for value-less attributes and comment order in `attribute-order` rule

### LINKED ISSUE #2692: `--fix` for `attribute-order` breaks `@tagName=""`
```hbs
<Foo class="asd" @tagName="" />
```

Running `--fix` on that leads to:

```hbs
<Foo @tagName class="asd" />
```

Which is broken.

### LINKED ISSUE #2693: `--fix` for `attribute-order` moves around comments
```hbs
<Foo
  class="asd asd qwe rty asd zxc asd qwe asd zxc asd qwe asd zxc qwe qwe qwe qwe qwe asd zxc qwe qwe qwe qwe qwe"
  {{! @glint-expect-error }}
  @foo={{1}} />
```

(the long `class` is for Prettier to not make the whole thing on one line)

Running `--fix` on that leads to:

```hbs
<Foo
  @foo={{1}}
  class="asd asd qwe rty asd zxc asd qwe asd zxc asd qwe asd zxc qwe qwe qwe qwe qwe asd zxc qwe qwe qwe qwe qwe"
  {{! @glint-expect-error }}
/>
```

Which is not nice because the comment is supposed to go with the `@foo` argument but it was moved at the end. This is a breaking change for `glint` directives. And just an annoyance for normal comments. :)

Not sure if that's a bug that's fixable though...

### PR DESCRIPTION
**Pull request by rob-long**
Fixes #2693
Fixes #2692 
- add additional test coverage for these fixes
- refactor for clarity
- add more inline docs

### CODE REVIEW

File: lib/rules/attribute-order.js [hunk 1]
```diff
@@ -453,45 +497,56 @@ export default class AttributeOrder extends Rule {
...
         if (orderA === orderB) {
           if (a.loc && b.loc) {
-            return a.loc.start.line * a.loc.start.column > b.loc.start.line * b.loc.start.column;
+            return this.calculateAttributeIndex(a) > this.calculateAttributeIndex(b);
           }
         }
         return orderA > orderB ? 1 : -1;
       });
 
+      for (const comment of node[AttributeType.COMMENTS]) {
```
Comment (bmish) [inline lib/rules/attribute-order.js:515]: Can you add a high-level sentence ahead of this loop to explain what it does?

File: lib/rules/attribute-order.js [hunk 2]
```diff
@@ -178,6 +193,33 @@ export default class AttributeOrder extends Rule {
       : AttributeType.ATTRIBUTES;
   }
 
+  getAttribute(nodeTokens, { type, name, value }) {
```
Comment (bmish) [inline lib/rules/attribute-order.js:196]: Can you add a high-level sentence ahead of each function to explain what it does?

File: test/unit/rules/attribute-order-test.js [hunk 3]
```diff
@@ -734,6 +734,79 @@ generateRuleTests({
...
+        `);
+      },
+    },
+    {
+      template: `<Foo
+      {{!-- @glint-expect-error --}}
+      id="op"
+      {{!-- @second --}}
+      @foo={{1}}
+    />`,
```
Comment (bmish) [inline test/unit/rules/attribute-order-test.js:777]: Can we also test with a double comment and with a trailing comment?

```
`<Foo
      {{!-- @glint-expect-error --}}
      id="op"
      {{!-- double comment --}}
      {{!-- @second --}}
      @foo={{1}}
      {{!-- trailing comment --}}
    />`

**Comment by bmish** [comment]
Note that you have to use the syntax `Fixes #123` in the PR description so this PR will automatically link and close the issues.

File: lib/rules/attribute-order.js (hunk 2 above)
Comment (rob-long) [reply lib/rules/attribute-order.js]: Refactored for clarity and added docs to most functions.

**Review by bmish — APPROVED** [review]
Thanks!

**Pull request merged**